### PR TITLE
added DateFormat constants with xmldocs

### PR DESCRIPTION
**Pull request by Mpdreamz**
fix for #2493

### CODE REVIEW

File: src/Nest/CommonOptions/DateFormat/DateFormat.cs [hunk 1]
```diff
@@ -0,0 +1,162 @@
+﻿namespace Nest.CommonOptions.DateFormat
```
Comment (russcam) [inline src/Nest/CommonOptions/DateFormat/DateFormat.cs]: namespace should be `"Nest"`

File: src/Nest/CommonOptions/DateFormat/DateFormat.cs [hunk 2]
```diff
@@ -0,0 +1,162 @@
...
+{
+	// ReSharper disable InconsistentNaming
+	//these const reflect their output on purpose
+	public static class DateFormat
+	{
+		///<summary>A formatter for the number of milliseconds since the epoch. Note, that this timestamp is subject to the limits of a Java Long.MIN_VALUE and Long.MAX_VALUE.</summary>
+		public const string epoch_millis = "epoch_mills";
```
Comment (russcam) [inline src/Nest/CommonOptions/DateFormat/DateFormat.cs]: Strong opinions about using the original names v.s. idiomatic .NET Pascal casing?

Comment (Mpdreamz) [inline src/Nest/CommonOptions/DateFormat/DateFormat.cs]: kinda :) `EpochMillis(seconds)` not that bad but turning `strict_t_time` into `StrictTTIme` :/ their is litttle guessing or ambiguity in keeping the casing which is a big plus over dogmatic casing, What do you  guys reckon?

Comment (gmarz) [inline src/Nest/CommonOptions/DateFormat/DateFormat.cs]: I'm cool with using the original names.

**Comment by Mpdreamz** [comment]
backported to `5.x` and `2.x`

**Pull request merged**